scipy/linalg/_matfuncs.py

```diff
@@ -250,9 +270,141 @@ def expm(A):
            [ 1.06860742+0.48905626j, -1.71075555+0.91406299j]])
 
     """
-    # Input checking and conversion is provided by sparse.linalg.expm().
-    import scipy.sparse.linalg
-    return scipy.sparse.linalg.expm(A)
+    a = np.asarray(A)
+    if a.size == 1 and a.ndim < 2:
+        return np.array([[np.exp(a.ravel()[0])]])
+
+    if a.ndim < 2:
+        raise LinAlgError('The input array must be at least two-dimensional')
+    if a.shape[-1] != a.shape[-2]:
+        raise LinAlgError('Last 2 dimensions of the array must be square')
+    n = a.shape[-1]
+    # Empty array
+    if min(*a.shape) == 0:
+        return np.empty_like(a)
```

Comment:
Hmm I didn't understand your point here but I was just trying to accommodate for empty inputs. Do I miss some edge case?

Reply:
Similar to the below case of a scalar, the original `expm` turns this case into arrays (as everything else). In your implementation `np.empty_like(a)` will return an empty matrix if the input was an empty matrix (rather than returning an empty array).

Reply:
Ah right. Indeed, that is the case but hopefully, it will scream somewhere else if the matrix is full. We have lots of checks for SciPy functions that we want to reduce and lean more towards the user to do the right thing. Especially `asarray` and `asanyarray` usage turned out to be not so useful in hindsight. Hence I am trying to limit myself from covering all possible edge cases but trying to keep the sunny-weather flow intact.

Reply:
Sure thing, I only pointed that out for potential backwards-compatibility concerns.